extension/indexer/successful_tx_subscriber.go

```diff
@@ -0,0 +1,39 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package indexer
+
+import (
+	"context"
+
+	"github.com/ava-labs/hypersdk/chain"
+)
+
+var _ AcceptedSubscriber = (*successfulTxSubscriber)(nil)
+
+type SuccessfulTxSubscriber interface {
+	Accepted(ctx context.Context, tx *chain.Transaction, result *chain.Result) error
+}
+
+// SuccessfulTxSubscriber calls the provided SucccessfulTxSubscriber for each successful transaction
+type successfulTxSubscriber struct {
+	successfulTxSubscriber SuccessfulTxSubscriber
+}
+
+func NewSuccessfulTxSubscriber(subscriber SuccessfulTxSubscriber) AcceptedSubscriber {
+	return &successfulTxSubscriber{subscriber}
+}
+
+func (s *successfulTxSubscriber) Accepted(ctx context.Context, blk *chain.StatelessBlock) error {
+	results := blk.Results()
+	for i, tx := range blk.Txs {
+		result := results[i]
+		if result.Success {
```

Comment:
I think this is how the current code behaves... but just wondering if we should perform this check or leave it to the implementation. We don't have to worry about it for this PR though.

Reply:
Discussed here https://github.com/ava-labs/hypersdk/pull/1143#discussion_r1679916304 and I opted to keep this, so that each VM only handles accepted actions